docs/metrics.md

```diff
@@ -58,67 +155,241 @@ at the current time, they should not be used to report historical data such as:
 Metrics are also ephemeral, meaning there is no guarantee that they will be persisted for any amount of time. If you need
 a way to view and analyze historical data, consider the [workflow archive](workflow-archive.md) or reporting to logs.
 
+### Counter, gauge and histogram
+
+The following terms have specific meanings for the metrics that we collect:
+
+- `counter`: A value that accumulates over time – you can think of this like an odometer on a car; it only ever goes up.
+    - The rate of change of counters can be a very powerful tool in understanding these metrics.
+- `gauge`: Measures a current value at the time it is read. An example would be the fuel gauge in a vehicle.
+- `histogram`: A client-side aggregation of values, such as request latencies. A histogram is a good choice if you are interested in value statistics. For example: How many requests take fewer than 1 second?
+
+Thanks to [OpenTelemetry](https://opentelemetry.io/docs/concepts/signals/metrics/#metric-instruments) for these explanations.
+
 ### Default Controller Metrics
 
 Metrics for the Four Golden Signals are:
 
-- Latency: `argo_workflows_queue_latency`
-- Traffic: `argo_workflows_count` and `argo_workflows_queue_depth_count`
-- Errors: `argo_workflows_count` and `argo_workflows_error_count`
-- Saturation: `argo_workflows_workers_busy` and `argo_workflows_workflow_condition`
+- Latency: `queue_latency`
+- Traffic: `gauge` and `queue_depth_gauge`
+- Errors: `count` and `error_count`
+- Saturation: `workers_busy` and `workflow_condition`
 
+!!! High cardinality
+    Some metric attributes may have high cardinality and are marked with ⚠️ to warn you. You may need to disable this metric or disable the attribute.
```

Comment:
```suggestion
    Some metric attributes may have high cardinality and are marked with ⚠️.
    You may want to disable the metric or attribute.
    
```

- We should use proper [`mkdocs-material` icons](https://squidfunk.github.io/mkdocs-material/reference/icons-emojis/) for this icon
- we should explain _why_ they may want to disable it, such as high storage usage or similar. Idk if there's a link we can provide for that, but that would be ideal
- Missing new line after note
- prefer 1 sentence per line of markdown, per [style guide](https://argo-workflows.readthedocs.io/en/release-3.5/doc-changes/)

Reply:
I don't see an precedent for usage of non-emoji based icons. I prefer the emoji based ones because all text editors render them as icons.
An explanation of high cardinality is mentioned at least twice elsewhere including a link to an explanation. I felt it was redundant to explain this more.